src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/RoomControllerTest.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.sandl.snlevents.controllers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.val;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.Room;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RoomService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(RoomController.class)
+public class RoomControllerTest {
+    public static final String URL = "/room";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private RoomService roomService;
+
+    @Test
+    public void fetchAllRooms_returnsRoomsFromService() throws Exception {
+        val rooms = createRooms();
+        when(roomService.getRooms()).thenReturn(createRooms());
+
+        val response = getMappedResponse(URL);
+        assertThat(response).isEqualTo(rooms);
+    }
+
+    private List<Room> createRooms() {
+        return new ArrayList<>(Arrays.asList(new Room()));
+    }
+
+    private Object getMappedResponse(String url) throws Exception {
```

Comment:
Possibly we could extract this method too, just make it more generic and add another parameter for type?